### ISSUE TITLE
Exception not thrown when creating non-Draft Invoice with incomplete LineItems

### ISSUE THREAD

**Issue by Healyhatman** [calcinai/xero-php#698]
When creating an invoice with status set to anything other than DRAFT, with incomplete line items (e.g. missing an account code, invalid tax rate), the response includes error messages but no exception is thrown. 

          "ValidationErrors" => array:2 [
            0 => array:1 [
              "Message" => "Account must be valid."
            ]
            1 => array:1 [
              "Message" => "Tax rate must be valid."
            ]
          ]

and

  -element_errors: array:1 [
    0 => "Tax rate must be valid"
  ]

**Comment by calcinai** [comment]
This will probably require a version bump if  we start throwing new exceptions.. Have you tested throwing them?

**Comment by Healyhatman** [comment]
No I haven't sorry, where does that happen?

**Comment by Healyhatman** [comment]
OK I've done some more testing - it throws the error as expected when using $invoice->save, but NOT when using $xero->saveAll([$invoice])

  -response_body: """
    <Response xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
      <Id>1f03b185-f8a2-4cca-b2e6-967b0b3dfa50</Id>
      <Status>OK</Status>
      <ProviderName>Care Master</ProviderName>
      <DateTimeUTC>2020-03-18T23:17:03.5291243Z</DateTimeUTC>
      <Invoices>
        <Invoice status="ERROR">
          <ValidationErrors>
            <ValidationError>
              <Message>Account code or ID must be specified</Message>
            </ValidationError>
          </ValidationErrors>
          <Contact>
            <ContactID>00000000-0000-0000-0000-000000000000</ContactID>
            <Name>Pete</Name>
          </Contact>
          <Date>2020-03-19T00:00:00</Date>
          <DueDate>2020-03-19T00:00:00</DueDate>
          <Status>AUTHORISED</Status>
          <LineAmountTypes>Exclusive</LineAmountTypes>
          <LineItems>
            <LineItem>
              <Description>test</Description>
              <UnitAmount>10.00</UnitAmount>
              <TaxType>OUTPUT</TaxType>
              <TaxAmount>1.00</TaxAmount>
              <LineAmount>10.00</LineAmount>
              <Quantity>1.0000</Quantity>
            </LineItem>
          </LineItems>
          <SubTotal>10.00</SubTotal>
          <TotalTax>1.00</TotalTax>
          <Total>11.00</Total>
          <CurrencyCode>AUD</CurrencyCode>
          <Type>ACCREC</Type>
          <InvoiceID>00000000-0000-0000-0000-000000000000</InvoiceID>
          <HasErrors>true</HasErrors>
        </Invoice>
      </Invoices>
    </Response>

**Comment by calcinai** [comment]
Yup, this is expected behaviour as it is adding the summerizeErrors flag.  saveAll can successfully save some and not others (per the Xero API), so you don't really want an exception in that case..

**Comment by Healyhatman** [comment]
OK so is there a method to get the errors? $response =
$xero->saveAll($theThings);  $response->getErrors() or somesuch? I'm very
sleepy after playing Doom Eternal til 4:30am

>

**Comment by Healyhatman** [comment]
```
// Save invoices. replace_data insures new contacts have their ContactIDs populated
$response = $xero->saveAll($invoices, true, true);

// Get errors
$errors = $response->getElementErrors();
```
